### PR TITLE
fix(mcdu): fix incorrect trip wind value handling on INIT-B page

### LINKED ISSUE #3745: [BUG] Incorrect entered Trip Wind value handling on INIT-B page
**Mod Version**

master

**Describe the bug**

Incorrect value entered to TRIP WIND placeholder when correctly entered without prefix. Incorrect values are also accepted.

**To Reproduce**

1. Go to INIT-B page
2. Enter any positive two-digit tail wind value without "+", "T", "TL" prefix. 12 for example.
3. Enter 4HD56

**Expected behavior**

2. TL012 appears in in the TRIP WIND placeholder
3. FORMAT ERROR displayed

**Actual behavior**

2. TL002 appears in in the TRIP WIND placeholder
3. TL005 appears in in the TRIP WIND placeholder

**References**

**Additional context**

Discord username (if different from GitHub): SergeyS#7450

### PR DESCRIPTION
**Pull request by SergeySoprykin**
Fixes #3745

## Summary of Changes

Fixed trip wind input string parsing. The problem caused by splitting input string by empty string separator. Also the format checks were insufficient.
Examples:
12 -> TL02
3HD45 -> HD045

## Additional context

Discord username (if different from GitHub): SergeyS#7450

## Testing instructions

Actions
1. Go to INIT-B page and enter tail winds: 12, +34, T56, TL78
2. Enter head winds: -12, H34, HD56
3. Enter incorrect values: 1HD23, 1TL23 and any other

Result
1. The values are correctly accepted and displayed values are: TL012, TL034, TL056, TL078
2. The values are correctly accepted and displayed values are: HD012, HD034, HD056
3. FORMAT ERROR displayed

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 1]
```diff
@@ -2461,30 +2461,22 @@ class FMCMainDisplay extends BaseAirliners {
...
+        const matchedIndex = validDelims.findIndex(element => s.startsWith(element));
+        const digits = matchedIndex >= 0 ? s.replace(validDelims[matchedIndex], "") : s;
+        const isNum = /^\d+$/.test(digits);
+        if (!isNum) {
             this.addNewMessage(NXSystemMessages.formatError);
             return false;
         }
+        const wind = parseInt(digits);
+        this._windDir = matchedIndex <= 2 ? this._windDirections.TAILWIND : this._windDirections.HEADWIND;
+        if (!this.isAvgWindInRange(wind)) {
```
Comment (beheh) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:2474]: This check could be inlined for now as it's not used in any other place.

**Review by beheh — APPROVED** [review]
One suggestion, but good for testing.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js (hunk 1 above)
Comment (SergeySoprykin) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js]: Removed the function. And also `wind < 0` check is not needed here. Thanks.

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 28.2.2021
Version of the sim: 1.13.16.0
PR Tested: #2746
Tier of testing: 1
Changes to observe: Trip wind input formats
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP 
2.) Turned on bat. and ext. power
3.) Played with trip wind input formats
 
Issues: -

Overall Rating: Good

Conclusions of the testing: Only formats are accepted like written in the description. Right error messages are shown for format errors or entries out of range.

Hope this helps :-)

**Comment by beheh** [comment]
Small change and tested by Q&A, merging.

**Comment by beheh** [comment]
Small change and tested by Q&A, ready to merge.

@SergeySoprykin Could you fix the changelog conflict? Then I'll merge this.

**Pull request merged**